src/appleseed/renderer/kernel/lighting/imagebasedlighting.cpp

```diff
@@ -204,9 +204,23 @@ void compute_ibl_environment_sampling(
 
         // Discard occluded samples.
         Spectrum transmission;
-        material_sampler.trace_simple(shading_context, incoming, transmission);
-        if (is_zero(transmission))
-            continue;
+        if (environment_edf.get_cast_shadows())
+        {
+            const ShadingPoint& shading_point = material_sampler.trace_full(
```

Comment:
Why did you change the logic here?

Reply:
Do you mean why I used trace_full instead of trace_simple? It's because trace_simple doesn't return the shading point.

Reply:
Why would you need the shading point? Why changing the logic (with a significant performance impact) when all is needed is wrapping the test in a "if cast shadows"?

Reply:
I'm sorry. I got confused. I was thinking about it a bit wrong. I'll update the PR.